shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -104,20 +105,38 @@
    *
    * <p>This must be called before any other native methods, and can be overridden by tests to avoid
    * loading native libraries.
+   *
+   * <p>This method should only be called once across all FlutterJNI instances.
    */
   public void loadLibrary() {
+    if (loadLibraryCalled) {
```

Comment:
Ya, though herein lies the cognitive ambiguity since it's not clear what methods the user is expected to call again on a second JNI and which ones are pointless no-ops or redundant. This doesn't really do anything except add docs and messages to point out that certain "static" functions don't need to be called again on a new JNI.